contracts/helpers/Rebalancer.sol

```diff
@@ -0,0 +1,152 @@
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+                                                 REBALANCING FUNCTIONS                                              
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////*/
+
+    /// @notice Swaps `tokenIn` for `tokenOut` through an intermediary agToken mint from `tokenIn` and
+    /// burn to `tokenOut`. Eventually, this transaction may be sponsored with an additional amount
+    /// of `tokenOut` based on the `orders` set by governance
+    /// @param amountIn Amount of `tokenIn` to bring for the rebalancing
+    /// @param amountOutMin Minimum amount of `tokenOut`that must be obtained from the swap
+    /// @param to Address to which `tokenOut` must be sent
+    /// @param deadline Timestamp before which this transaction must be included
+    /// @dev Contrarily to what is done in the Transmuter contract, here neither of `tokenIn` or `tokenOut`
+    /// should be an `agToken`
+    function swapExactInput(
+        uint256 amountIn,
+        uint256 amountOutMin,
+        address tokenIn,
+        address tokenOut,
+        address to,
+        uint256 deadline
+    ) external returns (uint256 amountOut) {
+        IERC20(tokenIn).safeTransferFrom(msg.sender, address(this), amountIn);
+        // Dealing with the allowance of the rebalancer to the Transmuter: this allowance is made infinite by default
+        uint256 allowance = IERC20(tokenIn).allowance(address(this), address(transmuter));
+        if (allowance < amountIn)
+            IERC20(tokenIn).safeIncreaseAllowance(address(transmuter), type(uint256).max - allowance);
+        // First, mint agToken from `tokenIn`
+        uint256 amountAgToken = transmuter.swapExactInput(
+            amountIn,
+            0,
+            tokenIn,
+            agToken,
+            address(this),
+            block.timestamp
+        );
+        // Then, burn the minted agToken to `tokenOut`
+        amountOut = transmuter.swapExactInput(amountAgToken, 0, agToken, tokenOut, to, deadline);
+        // Based on the `amountOut` obtained, checking whether this is eligible to a subsidy
+        uint256 subsidy = _computeSubsidy(tokenIn, tokenOut, amountOut);
+        if (subsidy > 0) {
+            // Everytime a subsidy takes place, the total subsidy budget must be reduced
+            orders[tokenIn][tokenOut].subsidyBudget -= subsidy;
+            budget[tokenOut] -= subsidy;
+            amountOut += subsidy;
+            IERC20(tokenOut).safeTransfer(to, subsidy);
+        }
+        if (amountOut < amountOutMin) revert TooSmallAmountOut();
+    }
+
+    /// @notice Simulates how much a call to `swapExactInput` with the same parameters would yield in terms
+    /// of `amountOut`
+    function quoteIn(uint256 amountIn, address tokenIn, address tokenOut) external view returns (uint256) {
```

Comment:
This computation is off as we don't consider the fees post swap

Reply:
Ah yes indeed, need to put a comment to explain that there might be a small divergence here

Reply:
And I don't want to handle the complexity of mimicking what'd be done here